rust_src/src/cmds.rs

```diff
@@ -214,20 +214,18 @@ pub fn delete_char(n: EmacsInt, killflag: bool) -> () {
 
     let buffer = ThreadState::current_buffer();
     let pos = buffer.pt + n as isize;
-    if !killflag {
-        if n < 0 {
-            if pos < buffer.begv {
-                xsignal!(Qbeginning_of_buffer);
-            } else {
-                unsafe { del_range(pos, buffer.pt) };
-            }
-        } else if pos > buffer.zv {
-            xsignal!(Qend_of_buffer);
+    if killflag {
+        call_raw!(Qkill_forward_chars, LispObject::from(n));
+    } else if n < 0 {
+        if pos < buffer.begv {
+            xsignal!(Qbeginning_of_buffer);
         } else {
-            unsafe { del_range(buffer.pt, pos) };
+            unsafe { del_range(pos, buffer.pt) };
         }
+    } else if pos > buffer.zv {
+        xsignal!(Qend_of_buffer);
     } else {
```

Comment:
Are you sure? `kill-forward-characters` might be subtly different.

Reply:
@nickdrozd my comment is on the code below this comment. Is `del_range` doing exactly what `kill-forward-chars` is doing?

Reply:
Before:
```rust
    if !killflag {
        if n < 0 {
            if pos < buffer.begv {
                xsignal!(Qbeginning_of_buffer);
            } else {
                unsafe { del_range(pos, buffer.pt) };
            }
        } else if pos > buffer.zv {
            xsignal!(Qend_of_buffer);
        } else {
            unsafe { del_range(buffer.pt, pos) };
        }
    } else {
        call_raw!(Qkill_forward_chars, LispObject::from(n));
    }
```

After:
```rust
    if killflag {
        call_raw!(Qkill_forward_chars, LispObject::from(n));
    } else if n < 0 {
        if pos < buffer.begv {
            xsignal!(Qbeginning_of_buffer);
        } else {
            unsafe { del_range(pos, buffer.pt) };
        }
    } else if pos > buffer.zv {
        xsignal!(Qend_of_buffer);
    } else {
        unsafe { del_range(buffer.pt, pos) };
    }
```

`kill-forward-chars` is called just when `killflag` is true, before and after. The major difference is that `if !killflag` is changed to `if killflag`, inverting the `if/else` blocks. But the logic should be the same.

The diff does a poor job of representing the changes in this case, because I guess it's trying to preserve continuity when there is none.

Reply:
!OH!

Thanks @nickdrozd. I did not look closely enough.